src/fwds_rvs_data.jl

```diff
@@ -860,15 +860,14 @@ tangent type. This method must be equivalent to `tangent_type(_typeof(primal))`.
 @foldable tangent_type(::Type{NoFData}, ::Type{R}) where {R<:IEEEFloat} = R
 @foldable tangent_type(::Type{F}, ::Type{NoRData}) where {F<:Array} = F
 
-# Union types. NOTE: Only `Union{Nothing, Array{<:Base.IEEEFloat}, Base.IEEEFloat}` are supported. 
 @foldable function tangent_type(
     ::Type{NoFData}, ::Type{R}
-) where {R<:Union{NoRData,T} where {T<:Base.IEEEFloat}}
+) where {R<:Union{NoRData,T} where {T<:Any}}
     return tangent_type(R)
 end
 @foldable function tangent_type(
     ::Type{F}, ::Type{NoRData}
-) where {F<:Union{NoFData,T} where {T<:Array{<:Base.IEEEFloat,N} where {N}}}
+) where {F<:Union{NoFData,T} where {T<:Any}}
```

Comment:
This should be enough for primal type `Union{Vector{Tuple{Int,Int,Vector{Tuple{Int,Int}}}},Nothing}` 

```suggestion
{F<:Union{NoFData,T} where {T<:Array{<:Any,N} where {, N}}}
```